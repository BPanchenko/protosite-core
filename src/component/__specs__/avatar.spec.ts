import AvatarElement from '../avatar';
import { configureToMatchImageSnapshot } from 'jest-image-snapshot';

expect.extend({
	toMatchImageSnapshot: configureToMatchImageSnapshot({
		comparisonMethod: 'ssim',
		failureThreshold: 0.1,
		failureThresholdType: 'percent'
	})
});
jest.setTimeout(60000);

describe('AvatarElement', () => {
	const avatar = new AvatarElement();

	it('Markup', () => {
		document.body.appendChild(avatar);
		expect(document.body).toMatchSnapshot();
	});

	describe('Visual Regression', () => {
		let containerHandle, image;

		beforeAll(async () => {
			const page = await global.browser.newBlankPage();

			containerHandle = await page.$('#container');
			await page.evaluate((container) => {
				const avatar = document.createElement('c-avatar');
				container.appendChild(avatar);
			}, containerHandle);

			image = await page.screenshot({ fullPage: true });
		});

		afterAll(async () => {
			await containerHandle.dispose();
		});

		it('horizontal', async () => {
			expect(image).toMatchImageSnapshot({
				diffDirection: 'horizontal'
			});
		});

		it('vertical', async () => {
			expect(image).toMatchImageSnapshot({
				diffDirection: 'vertical'
			});
		});
	});
});
