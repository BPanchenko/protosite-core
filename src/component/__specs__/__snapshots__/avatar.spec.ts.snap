// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarElement Markup 1`] = `
<body>
  <c-avatar
    class="u-display-contents"
  />
</body>
`;
